c10/core/StorageImpl.h

```diff
@@ -224,11 +225,22 @@ struct C10_API StorageImpl final : public c10::intrusive_ptr_target {
     resizable_ = false;
   }
 
+  void set_received_cuda(bool received_cuda) {
```

Comment:
Would be good to note that this method is expected be used purely right after you constructed a storage, and not to modify the status later in its lifetime